kviz/dense.py

```diff
@@ -78,60 +48,71 @@ class DenseGraph():
+                Set the marker of the points in the pyplot graph.
+
+            customize_graph:
+                Changes the colors and shapes of the graph.
+                It can be called after a DenseGraph instance is created
+                and before render().
+
             render :
                 Can print the network architecture or, if input
                 is provided, show a GIF of the activations of each
                 Neuron based on the input provided.
 
     """
 
-    def __init__(self, model, input_color='#3498db', inner_color='#2ecc71', output_color='#3498db',
-                 edge_clr='#B20000', input_shape='circle', inner_shape='circle', output_shape='circle'):
+    def __init__(self, model):
         """
 
             Parameters:
                 model: A keras model.
                     Since our class is called DenseGraph(), the keras model should only contain dense layers.
-                input_color: str.
-                    The color of the input layer in hex form (e.g. "#FFFFFF").
-                inner_color: str.
-                    The color of the inner layer(s) in hex form.
-                output_color: str.
-                    The color of the output layer in hex form.
-                edge_clr: str.
-                    The color of the edge connecting nodes of different layer. In hex form.
-                input_shape: str.
-                    The shape of the nodes in the input layer. Should be a valid shape (e.g. "polygon").
-                    Otherwise the shape will be "box" for invalid input.
-                    Check https://graphviz.org/doc/info/shapes.html for some valid shapes.
-                inner_shape: str.
-                    The shape of the nodes in the inner layer(s). Should be a valid shape.
-                output_shape:
-                    The shape of the nodes in the output layer. Should be a valid shape.
+
         """
-        expression = r'^#(?:[0-9a-fA-F]{3}){1,2}$'
-        if re.search(expression, input_color) and re.search(expression, inner_color) and \
-                re.search(expression, output_color) and re.search(expression, edge_clr):
-            self.input_layer_node_color = input_color
-            self.inner_layer_node_color = inner_color
-            self.output_layer_node_color = output_color
-            self.edge_color = edge_clr
-            self.input_layer_node_shape = input_shape
-            self.inner_layer_node_shape = inner_shape
-            self.output_layer_node_shape = output_shape
-        else:
-            self.input_layer_node_color = '#3498db'
-            self.inner_layer_node_color = '#2ecc71'
-            self.output_layer_node_color = '#3498db'
-            self.edge_color = '#B20000'
-            self.input_layer_node_shape = 'circle'
-            self.inner_layer_node_shape = 'circle'
-            self.output_layer_node_shape = 'circle'
+        # the color&shape for _snap_X()
+        self.x_color = "#3498db"
```

Comment:
this is only useful for animate_learning right? is there a reason it's a global attribute and not a function parameter?

Reply:
They will be changed to function parameters in `render` and `_snap_X`